src/gateways/createArchiveLoader.ts

```diff
@@ -44,12 +44,13 @@ const defaultOptions = () => ({
   pageCache: createCache<string, ArchivePage>({maxRecords: 20}),
 });
 
-export default (archivePath: string, options: Options = {}) => {
+export default (getArchivePath: () => string, options: Options = {}) => {
   const {pageCache, bookCache, appPrefix, archivePrefix, disablePerBookPinning} = {
     ...defaultOptions(),
     ...options,
   };
 
+  const archivePath = getArchivePath();
```

Comment:
this part of the function is run when the gateway is created, so you're still not picking up changes in the cached value over time